examples/alife-boid/src/boid.rs

```diff
@@ -0,0 +1,117 @@
+use crate::rand::Rand;
+use crate::vector::Vector;
+use std::f64::consts::PI;
+
+const HEIGHT: f64 = 400.0;
+const WIDTH: f64 = 600.0;
+const VELOCITY_SIZE: f64 = 5.0;
+const ALIGNMENT_RADIOUS: f64 = 100.0;
+const ALIGNMENT_WEIGHT: f64 = 3.0;
+const COHENSION_RADIOUS: f64 = 200.0;
+const COHENSION_WEIGHT: f64 = 1.0;
+const SEPARATION_RADIOUS: f64 = 50.0;
+const SEPARATION_WEIGHT: f64 = 1.0;
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct Boid {
+    pub position: Vector,
+    pub velocity: Vector,
+}
+
+impl Boid {
+    pub fn new(rng: &mut Rand) -> Boid {
+        let theta = rng.next_f64() * PI * 2.0;
+        Boid {
+            position: Vector::new(WIDTH * rng.next_f64(), HEIGHT * rng.next_f64()),
+            velocity: Vector::new(theta.cos() * VELOCITY_SIZE, theta.sin() * VELOCITY_SIZE),
+        }
+    }
+
+    fn calc_alignment(&self, boids: &[Boid]) -> Vector {
```

Comment:
This can be optimized the same way as `calc_cohesion`; by calculating the velocity sum beforehand.